### PR TITLE
Add ruff

### PR DESCRIPTION
**Pull request by henrikstranneheim**
## Description

Add Ruff to dev packages installation and pre-commit. 

### Added

- Ruff package

### Changed

- Apply Ruff suggestions to entire code base

### Fixed

- Remove unused import
- Remove implicit imports
- Add models to analyis type hint (will correct import in a later PR)

### How to prepare for test
- [ ] ssh to hasta (depending on type of change)
- [ ] activate stage: `us`
- [ ] request trailblazer-stage on hasta: `paxa`
- [ ] install on stage:
    ```Shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_trailblazer -t trailblazer -b add_ruff -a
    ```
- [ ] ssh to clinical-db (depending on type of change)
- [ ] install on stage:
`bash /home/proj/production/servers/resources/clinical-db.scilifelab.se/update-trailblazer-ui-stage.sh [THIS-BRANCH-NAME]`

### How to test
- [ ] login to ...
- [ ] do ...

### Expected test outcome
- [ ] check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review
- [ ] tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: trailblazer/store/api.py [hunk 1]
```diff
@@ -257,19 +257,19 @@ def set_analysis_completed(self, analysis_id: int) -> None:
 
     def set_analysis_uploaded(self, case_id: str, uploaded_at: dt.datetime) -> None:
         """Setting analysis uploaded at."""
-        analysis_obj: Analysis = self.get_latest_analysis(case_id=case_id)
+        analysis_obj: models.Analysis = self.get_latest_analysis(case_id=case_id)
```
Comment (seallard) [inline trailblazer/store/api.py:260]: Do we want to use these types of imports?

**Review by seallard — APPROVED** [review]
Nice to try out ruff 👍

**Comment by henrikstranneheim** [comment]
### trailblazer  update-analysis 271866 ✅ 
```
[henrik.stranneheim@hasta:~] [S_base] 2m11s $ trailblazer  update-analysis 271866
Status in SLURM: civilsole - 271866
         id                                         step     status  time_limit  time_elapsed              started
0   4798531                               vep_somatic_sv  COMPLETED    12:00:00            27  2023-06-03T21:24:38
1   4798526                     vep_somatic_clinical_snv  COMPLETED    18:00:00             1  2023-06-03T21:20:51
2   4798524                     vep_somatic_research_snv  COMPLETED    18:00:00            43  2023-06-03T20:37:45
3   4798512                            sentieon_DNAscope  COMPLETED  1-00:00:00            18  2023-06-03T19:43:16
4   4798515                  sentieon_TNscope_tumor_only  COMPLETED  1-00:00:00            52  2023-06-03T19:43:16
5   4798510                    sentieon_base_calibration  COMPLETED  1-00:00:00            14  2023-06-03T19:28:45
6   4798506                               mergeBam_tumor  COMPLETED  1-06:00:00           313  2023-06-03T14:15:09
7   4798504                             sentieon_realign  COMPLETED  1-00:00:00             7  2023-06-03T14:07:16
8   4798501                               sentieon_dedup  COMPLETED  1-00:00:00             7  2023-06-03T13:59:21
9   4798500                          sentieon_align_sort  COMPLETED  1-00:00:00            84  2023-06-03T12:34:55
10  4798535                  bcftools_filter_sv_clinical  COMPLETED    12:00:00             1  2023-06-03T21:56:10
11  4798533     annotate_clinical_observation_somatic_sv  COMPLETED    12:00:00             1  2023-06-03T21:54:20
12  4798534                  bcftools_filter_sv_research  COMPLETED    12:00:00             1  2023-06-03T21:54:20
13  4798532         annotate_swegen_frequency_somatic_sv  COMPLETED    12:00:00             2  2023-06-03T21:52:02
14  4798527                 bcftools_quality_filter_svdb  COMPLETED    12:00:00             0  2023-06-03T21:24:35
15  4798520              vcf2cytosure_convert_tumor_only  COMPLETED    12:00:00             0  2023-06-03T21:24:09
16  4798525                        svdb_merge_tumor_only  COMPLETED     2:00:00             0  2023-06-03T21:24:09
17  4798517                      bcftools_process_SV_CNV  COMPLETED     1:00:00             0  2023-06-03T21:24:05
18  4798530  bcftools_filter_tnscope_clinical_tumor_only  COMPLETED     3:00:00             0  2023-06-03T21:21:56
19  4798528                              tmb_calculation  COMPLETED    12:00:00            25  2023-06-03T21:20:51
20  4798529  bcftools_filter_tnscope_research_tumor_only  COMPLETED     4:00:00             1  2023-06-03T21:20:51
21  4798521   bcftools_quality_filter_tnscope_tumor_only  COMPLETED    12:00:00             0  2023-06-03T20:37:05
22  4798522                          cnvpytor_tumor_only  COMPLETED     5:00:00            27  2023-06-03T20:37:05
23  4798519                  bcftools_view_split_variant  COMPLETED    12:00:00             1  2023-06-03T20:35:55
24  4798513                         delly_cnv_tumor_only  COMPLETED    12:00:00            52  2023-06-03T20:31:33
25  4798518                           vep_germline_tumor  COMPLETED    12:00:00            71  2023-06-03T20:01:40
26  4798516                                      multiqc  COMPLETED       15:00             0  2023-06-03T19:43:16
27  4798514                           vep_germline_tumor  COMPLETED    12:00:00            15  2023-06-03T19:42:41
28  4798507                          delly_sv_tumor_only  COMPLETED    12:00:00            62  2023-06-03T19:28:45
29  4798508                         tiddit_sv_tumor_only  COMPLETED    10:00:00            45  2023-06-03T19:28:45
30  4798509                               manta_germline  COMPLETED     5:00:00            13  2023-06-03T19:28:45
31  4798511                             manta_tumor_only  COMPLETED    10:00:00            15  2023-06-03T19:28:45
32  4798502                     picard_CollectWgsMetrics  COMPLETED    12:00:00           127  2023-06-03T14:07:16
33  4798503                         sentieon_wgs_metrics  COMPLETED    12:00:00            35  2023-06-03T14:07:16
34  4798505                picard_CollectMultipleMetrics  COMPLETED    12:00:00            75  2023-06-03T14:07:16
35  4798499                                        fastp  COMPLETED    12:00:00            22  2023-06-03T12:12:22
36  4798497                                    fastp_umi  COMPLETED  1-00:00:00            27  2023-06-03T11:44:30
37  4798498                                       fastqc  COMPLETED    12:00:00            92  2023-06-03T11:44:30
38  4798536                    collect_custom_qc_metrics  COMPLETED       15:00             0  2023-06-03T21:57:56
39  4798523                                   cnv_report  COMPLETED       15:00             0  2023-06-03T21:04:37
40  4798537                                          all     FAILED       15:00             0  2023-06-03T21:58:01
Updated status civilsole - 271866: failed
```

File: trailblazer/store/api.py (hunk 1 above)
Comment (henrikstranneheim) [reply trailblazer/store/api.py:260]: Fixed here: https://github.com/Clinical-Genomics/trailblazer/pull/209

**Comment by henrikstranneheim** [comment]
Deployed on Hasta:

```
done.
Log deploy... done.
trailblazer, version 12.0.6
[henrik.stranneheim@hasta:~] [P_base] 2m1s $
```

**Comment by henrikstranneheim** [comment]
Automatic deploy action failed on cg-prod.services for unclear reasons

Manually fixed by:

```
[henrik.stranneheim@hasta:~] [base] $ ssh cg-prod-services.scilifelab.se
Last login: Wed May 24 09:12:16 2023 from 130.237.81.137
[henrik.stranneheim@cg-prod-services ~]$ sudo -iu hiseq.clinical
[hiseq.clinical@cg-prod-services ~]$ ssh localhost
Last login: Wed May 31 07:33:23 2023 from ::1
[hiseq.clinical@cg-prod-services ~]$ podman ps
CONTAINER ID  IMAGE                                                     COMMAND               CREATED         STATUS             PORTS                   NAMES
e010dc37d7af  docker.io/clinicalgenomics/cg-deployment:latest           /bin/sh -c gunico...  2 months ago    Up 2 months ago    0.0.0.0:7082->8000/tcp  cg_deployment-latest
1bf0fb33ca6a  docker.io/clinicalgenomics/arnold:latest                  /bin/sh -c gunico...  2 weeks ago     Up 2 weeks ago     0.0.0.0:7080->8000/tcp  arnold-latest
8aa2ab8d42f1  docker.io/clinicalgenomics/preclinvar:latest              /bin/sh -c gunico...  2 weeks ago     Up 2 weeks ago     0.0.0.0:7099->8000/tcp  preclinvar-latest
45c94dd60128  docker.io/clinicalgenomics/genotype-api:latest            /bin/sh -c gunico...  2 weeks ago     Up 2 weeks ago     0.0.0.0:7098->8000/tcp  genotype-external-latest
da56cbf379dc  docker.io/clinicalgenomics/scout-reviewer-service:latest                        2 weeks ago     Up 2 weeks ago     0.0.0.0:7103->5050/tcp  scout_reviewer_service-latest
bfa016237f19  docker.io/clinicalgenomics/cgbeacon2-server:latest        /bin/sh -c gunico...  2 weeks ago     Up 2 weeks ago     0.0.0.0:7091->8000/tcp  beacon-latest
de40efe9fb1c  docker.io/clinicalgenomics/loqusdbapi:latest              /bin/sh -c gunico...  2 weeks ago     Up 2 weeks ago     0.0.0.0:7079->8000/tcp  loqusdbapi-rd-latest
a4eb582c4124  docker.io/clinicalgenomics/genotype:latest                /bin/sh -c gunico...  2 weeks ago     Up 2 weeks ago     0.0.0.0:7096->8000/tcp  genotype-v1-latest
a25d6d552b2f  docker.io/clinicalgenomics/vogue:latest                   /bin/sh -c gunico...  2 weeks ago     Up 2 weeks ago     0.0.0.0:7075->8000/tcp  vogue-latest
0d3cafe545a0  docker.io/clinicalgenomics/genotype-api:latest            /bin/sh -c gunico...  2 weeks ago     Up 2 weeks ago     0.0.0.0:7097->8000/tcp  genotype-internal-latest
c58bad2a8bde  docker.io/clinicalgenomics/sendmail-container:latest      /bin/sh -c gunico...  2 weeks ago     Up 2 weeks ago                             sendmail-latest
ba311d124dab  docker.io/clinicalgenomics/trailblazer:latest             /bin/sh -c gunico...  2 weeks ago     Up 2 weeks ago     0.0.0.0:7076->8000/tcp  trailblazer-latest
37bd55e10acb  docker.io/clinicalgenomics/statina:latest                 /bin/sh -c gunico...  2 weeks ago     Up 2 weeks ago     0.0.0.0:7077->8000/tcp  statina-latest
4e7d33f58c56  docker.io/clinicalgenomics/patientmatcher-server:latest   /bin/sh -c gunico...  2 weeks ago     Up 2 weeks ago     0.0.0.0:7090->8000/tcp  patientmatcher_server-latest
39096f0c52d7  docker.io/clinicalgenomics/scout-server:v4.68             /bin/sh -c gunico...  2 weeks ago     Up 2 weeks ago     0.0.0.0:7073->8000/tcp  scout-v4.68
49a137839cb5  docker.io/clinicalgenomics/gens:latest                    /bin/sh -c gunico...  2 weeks ago     Up 2 weeks ago     0.0.0.0:7470->5000/tcp  gens-latest
cb5be4b4e536  docker.io/clinicalgenomics/cg:v33.2.0                     /bin/sh -c gunico...  5 days ago      Up 5 days ago      0.0.0.0:7777->8000/tcp  cg-v33.2.0
d342c1b9ac7e  docker.io/clinicalgenomics/atlas-api:9.0.2                /bin/sh -c gunico...  18 minutes ago  Up 18 minutes ago  0.0.0.0:7081->8000/tcp  atlas-api-9.0.2
511c6259c129  docker.io/clinicalgenomics/atlas:9.0.2                                          18 minutes ago  Up 18 minutes ago  0.0.0.0:7083->8080/tcp  atlas-doc-9.0.2
[hiseq.clinical@cg-prod-services ~]$
```

```
[hiseq.clinical@cg-prod-services ~]$ systemctl --user stop trailblazer.target
[hiseq.clinical@cg-prod-services ~]$ systemctl --user start trailblazer@latest.service
[hiseq.clinical@cg-prod-services ~]$ systemctl --user status trailblazer@latest.service
```

```
[hiseq.clinical@cg-prod-services ~]$ podman ps
CONTAINER ID  IMAGE                                                     COMMAND               CREATED         STATUS             PORTS                   NAMES
e010dc37d7af  docker.io/clinicalgenomics/cg-deployment:latest           /bin/sh -c gunico...  2 months ago    Up 2 months ago    0.0.0.0:7082->8000/tcp  cg_deployment-latest
1bf0fb33ca6a  docker.io/clinicalgenomics/arnold:latest                  /bin/sh -c gunico...  2 weeks ago     Up 2 weeks ago     0.0.0.0:7080->8000/tcp  arnold-latest
8aa2ab8d42f1  docker.io/clinicalgenomics/preclinvar:latest              /bin/sh -c gunico...  2 weeks ago     Up 2 weeks ago     0.0.0.0:7099->8000/tcp  preclinvar-latest
45c94dd60128  docker.io/clinicalgenomics/genotype-api:latest            /bin/sh -c gunico...  2 weeks ago     Up 2 weeks ago     0.0.0.0:7098->8000/tcp  genotype-external-latest
da56cbf379dc  docker.io/clinicalgenomics/scout-reviewer-service:latest                        2 weeks ago     Up 2 weeks ago     0.0.0.0:7103->5050/tcp  scout_reviewer_service-latest
bfa016237f19  docker.io/clinicalgenomics/cgbeacon2-server:latest        /bin/sh -c gunico...  2 weeks ago     Up 2 weeks ago     0.0.0.0:7091->8000/tcp  beacon-latest
de40efe9fb1c  docker.io/clinicalgenomics/loqusdbapi:latest              /bin/sh -c gunico...  2 weeks ago     Up 2 weeks ago     0.0.0.0:7079->8000/tcp  loqusdbapi-rd-latest
a4eb582c4124  docker.io/clinicalgenomics/genotype:latest                /bin/sh -c gunico...  2 weeks ago     Up 2 weeks ago     0.0.0.0:7096->8000/tcp  genotype-v1-latest
a25d6d552b2f  docker.io/clinicalgenomics/vogue:latest                   /bin/sh -c gunico...  2 weeks ago     Up 2 weeks ago     0.0.0.0:7075->8000/tcp  vogue-latest
0d3cafe545a0  docker.io/clinicalgenomics/genotype-api:latest            /bin/sh -c gunico...  2 weeks ago     Up 2 weeks ago     0.0.0.0:7097->8000/tcp  genotype-internal-latest
c58bad2a8bde  docker.io/clinicalgenomics/sendmail-container:latest      /bin/sh -c gunico...  2 weeks ago     Up 2 weeks ago                             sendmail-latest
37bd55e10acb  docker.io/clinicalgenomics/statina:latest                 /bin/sh -c gunico...  2 weeks ago     Up 2 weeks ago     0.0.0.0:7077->8000/tcp  statina-latest
4e7d33f58c56  docker.io/clinicalgenomics/patientmatcher-server:latest   /bin/sh -c gunico...  2 weeks ago     Up 2 weeks ago     0.0.0.0:7090->8000/tcp  patientmatcher_server-latest
39096f0c52d7  docker.io/clinicalgenomics/scout-server:v4.68             /bin/sh -c gunico...  2 weeks ago     Up 2 weeks ago     0.0.0.0:7073->8000/tcp  scout-v4.68
49a137839cb5  docker.io/clinicalgenomics/gens:latest                    /bin/sh -c gunico...  2 weeks ago     Up 2 weeks ago     0.0.0.0:7470->5000/tcp  gens-latest
cb5be4b4e536  docker.io/clinicalgenomics/cg:v33.2.0                     /bin/sh -c gunico...  5 days ago      Up 5 days ago      0.0.0.0:7777->8000/tcp  cg-v33.2.0
d342c1b9ac7e  docker.io/clinicalgenomics/atlas-api:9.0.2                /bin/sh -c gunico...  38 minutes ago  Up 38 minutes ago  0.0.0.0:7081->8000/tcp  atlas-api-9.0.2
511c6259c129  docker.io/clinicalgenomics/atlas:9.0.2                                          38 minutes ago  Up 38 minutes ago  0.0.0.0:7083->8080/tcp  atlas-doc-9.0.2
d1ecc9ebc157  docker.io/clinicalgenomics/trailblazer:latest             /bin/sh -c gunico...  45 seconds ago  Up 45 seconds ago  0.0.0.0:7076->8000/tcp  trailblazer-latest
```

After manually restarting the container - the automatic deployment action to prod  was re-runned and then it worked.

**Pull request merged**